### PR TITLE
Doc: Automatically generate the members of each module

### PR DESCRIPTION
**Pull request by xuhdev**
---

## Checklist:

<details open>

<summary></summary>

- Does this pull request close an issue? We encourage you to open an issue first if this pull request (PR) is not a
  minor change.
  - [ ] No
    - [ ] The change in this pull request is minor.
  - [x] Yes
    - Close #{internal}

For the following questions, only check the boxes that are applicable.

- [x] Ensure that the pull request text above the horizontal line is descriptive.
- [x] Add/update relevant tests.
- [x] Add/update relevant documents.

- Do you have triage permission of this repository?
  - [ ] No
    - You are good.
  - [x] Yes
    - Does this pull request close an issue?
      - [ ] No, this pull request also acts as an issue by itself.
        - [ ] Assign yourself.
        - [ ] Label this pull request.
        - [ ] Put a milestone.
        - [ ] Put this pull request under "In Progress" or "Under Review" pipeline on ZenHub.
        - [ ] Put this pull request under the appropriate epic on ZenHub.
        - [ ] Put an estimate on ZenHub.
        - [ ] Add dependency relationship with other issues/pull requests on ZenHub.
      - [x] Yes, this pull request addresses an issue and does not act as an issue.
        - [x] Connect this pull request to the underlying issue on ZenHub.
        - [x] DON'T:
          - DON'T assign yourself or anyone else.
          - DON'T label.
          - DON'T put a milestone.
          - DON'T put this pull request under any epic on ZenHub.
          - DON'T put an estimate on ZenHub.
    - [x] When the pull request is ready, request review.
</details>

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -12,12 +12,31 @@ PyDAX is a simple Python API for downloading and loading datasets from IBM's `Da
 
    user_guide/*
 
-.. toctree::
-   :caption: API Reference
-   :glob:
-   :maxdepth: 1
+API references
```
Comment (edwardleardi) [inline docs/source/index.rst]:
```suggestion
API References
```
Heading consistency

File: docs/source/index.rst [hunk 2]
```diff
@@ -12,12 +12,31 @@ PyDAX is a simple Python API for downloading and loading datasets from IBM's `Da
...
+--------------
+
+.. autosummary::
+   :caption: API references
```
Comment (edwardleardi) [inline docs/source/index.rst]:
```suggestion
   :caption: API References
```

File: docs/source/index.rst [hunk 3]
```diff
@@ -12,12 +12,31 @@ PyDAX is a simple Python API for downloading and loading datasets from IBM's `Da
...
+   pydax
+   pydax.dataset
+   pydax.exceptions
+   pydax.schema
+
+Loaders
+~~~~~~~
+
+.. autosummary::
+   :caption: API references (Loaders)
```
Comment (edwardleardi) [inline docs/source/index.rst]:
```suggestion
   :caption: API References (Loaders)
```

**Review by edwardleardi — CHANGES_REQUESTED** [review]
LGTM, thanks for figuring this out. The sidebar might get a bit messy in case we add more modules structured like `loaders` in the future, but it shouldn't be too big a deal.

**Comment by xuhdev** [comment]
> LGTM, thanks for figuring this out. The sidebar might get a bit messy in case we add more modules structured like `loaders` in the future, but it shouldn't be too big a deal.

We can always submerge `loaders` when it starts becoming messy.

**Review by edwardleardi — APPROVED** [review]

**Pull request merged**